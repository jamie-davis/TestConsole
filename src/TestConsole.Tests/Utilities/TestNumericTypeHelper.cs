using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using NUnit.Framework;
using TestConsoleLib;
using TestConsoleLib.Testing;
using TestConsoleLib.Utilities;

namespace TestConsole.Tests.Utilities
{
    [TestFixture]
    public class TestNumericTypeHelper
    {
        private readonly static Type[] NumericTypes = 
        {
            typeof(sbyte),
            typeof(byte),
            typeof(short),
            typeof(ushort),
            typeof(int),
            typeof(uint),
            typeof(long),
            typeof(ulong),
            typeof(decimal),
            typeof(float),
            typeof(double)
        };

        private readonly static Type[] NonNumericTypes = 
        {
            typeof(string),
            typeof(char),
            typeof(DateTime),
            typeof(TimeSpan),
            typeof(bool),
            typeof(object),
        };

        [Test]
        public void NumericTypesAreIdentified()
        {
            //Arrange
            var output = new Output();
            
            //Act
            output.WrapLine("Numeric types");
            output.WriteLine();
            output.FormatTable(NumericTypes.Select(t => new { Type = t.Name, IsNumeric = NumericTypeHelper.IsNumeric(t) }));
            output.WriteLine();
            output.WriteLine();
            output.WriteLine();
            output.WrapLine("Non-Numeric types");
            output.WriteLine();
            output.FormatTable(NonNumericTypes.Select(t => new { Type = t.Name, IsNumeric = NumericTypeHelper.IsNumeric(t) }));

            //Assert
            output.Report.Verify();
        }
    }
}
