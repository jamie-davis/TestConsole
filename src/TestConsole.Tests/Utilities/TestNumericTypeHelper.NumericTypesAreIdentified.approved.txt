Numeric types                                                                                  
                                                                                               
        Is                                                                                     
Type    Numeric                                                                                
------- -------                                                                                
SByte   True                                                                                   
Byte    True                                                                                   
Int16   True                                                                                   
UInt16  True                                                                                   
Int32   True                                                                                   
UInt32  True                                                                                   
Int64   True                                                                                   
UInt64  True                                                                                   
Decimal True                                                                                   
Single  True                                                                                   
Double  True                                                                                   
                                                                                               
                                                                                               
                                                                                               
Non-Numeric types                                                                              
                                                                                               
         Is                                                                                    
Type     Numeric                                                                               
-------- -------                                                                               
String   False                                                                                 
Char     False                                                                                 
DateTime False                                                                                 
TimeSpan False                                                                                 
Boolean  False                                                                                 
Object   False                                                                                 
                                                                                               